### ISSUE TITLE
Garbage collection doesn't delete images tagged with multiple repositories

### ISSUE THREAD

**Issue by chbatey** [kubernetes/kubernetes#28491]
This leads to our clusters slowly filling up all available space.

For our cluster we have the same image tagged with multiple repositories e.g. registry/test/name and then registry/release-candidate/name

The garbage collector only deletes images by id and docker (or at least the version we're using 1.8.3) doesn't support deleting images by id when there are multiple tags.

```
Jul 04 16:16:59 ip-10-50-185-124.internal start-kubelet.sh[2485]: I0704 16:16:59.184024    2485 image_manager.go:291] [ImageManager]: Removing image "dc1e6c5406648977b738cda24f4058fc6e6e3587da90f4a5945137f206f4f458" to free 711948567 bytes
Jul 04 16:16:59 ip-10-50-185-124.internal docker[2444]: time="2016-07-04T16:16:59.183803091Z" level=error msg="HTTP Error" err="Conflict, cannot delete image b76d6aaf0507dbc6c5cf47ebadac33de6df577de2b88d159f2990346ae42da41 because it is tagged in multiple repositories, use -f to force" statusCode=409
```

Could we also delete via repository name and then by id?

If so happy to work on this and raise a PR.

**Comment by chbatey** [comment]
The spotify docker gc has a similar option but just uses -f on the image removal: https://github.com/spotify/docker-gc

**Comment by yujuhong** [comment]
Thanks for reporting the issue! I believe this is fixed in #29316.
